.github/workflows/deploy.yml

```diff
@@ -28,20 +33,26 @@ jobs:
       id: buildx
       uses: docker/setup-buildx-action@v1
 
-    - name: Build and push base image
+    - name: Build and push base image (${{ matrix.dockerbase }})
       id: docker_build
       uses: docker/build-push-action@v2
       with:
-        context: ./docker/base
+        context: ./docker/base/${{ matrix.dockerbase }}
         push: true
         tags: |
-          exaworks/sdk-base:latest
+          exaworks/sdk-base:${{ matrix.dockerbase }}
         platforms: linux/amd64
 
-    - name: Build and push composite image
+    - name: Build and push composite image (${{ matrix.dockerbase }})
       run: |
-        docker build -t rp docker/rp
+        docker build -t rp docker/rp --build-arg BASE_IMAGE=exaworks/sdk-base:${{ matrix.dockerbase }}
         docker build -t rp_parsl --build-arg BASE_IMAGE=rp docker/Parsl
         docker build -t rp_parsl_swift --build-arg BASE_IMAGE=rp_parsl docker/swift-t
         docker build -t exaworks/sdk --build-arg BASE_IMAGE=rp_parsl_swift docker/flux
```

Comment:
we still need to add tag (os platform) for each image, right?
```
        docker build \
          -t rp:${{ matrix.dockerbase }} \
          --build-arg BASE_IMAGE=exaworks/sdk-base:${{ matrix.dockerbase }} \
          docker/rp
        docker build \
          -t rp_parsl:${{ matrix.dockerbase }} \
          --build-arg BASE_IMAGE=rp:${{ matrix.dockerbase }} \
          docker/Parsl
        docker build \
          -t rp_parsl_swift:${{ matrix.dockerbase }} \
          --build-arg BASE_IMAGE=rp_parsl:${{ matrix.dockerbase }} \
          docker/swift-t
        docker build \
          -t rp_parsl_swift_flux:${{ matrix.dockerbase }} \
          --build-arg BASE_IMAGE=rp_parsl_swift:${{ matrix.dockerbase }} \
          docker/flux
```
and the last line is actually push back to the base image?

Reply:
Good idea.  I don't think it is strictly necessary since each builder runs in its own VM, but it won't hurt to add them and explicit is always better than implicit.

> and the last line is actually push back to the base image?

Subtle difference but it pushes to `exaworks/sdk` (which contains all 4 components) as opposed to `exaworks/sdk-base` (which just contains dependencies).